Warnings about jring

#649 Fixes #637 by storing `UnifiedTask*` in jring.

However, I found the warning:

```
/usr/include/x86_64-linux-gnu/bits/string_fortified.h:29:33: warning: ‘void* __builtin___memcpy_chk(void*, const void*, long unsigned int, long unsigned int)’ writing 16 bytes into a region of size 8 overflows the destination [-Wstringop-overflow=]
   29 |   return __builtin___memcpy_chk (__dest, __src, __len,
      |          ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
   30 |                                  __glibc_objsize0 (__dest));
      |                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~
```

The reason is that `jring` casts the destination buffer to `j_int128_t*` and uses memcpy to copy **16 byte**s for efficiency, while `UnifiedTask*` is **8 bytes**. Basically, the warning is harmless because only the first 8 bytes (the pointer value) are actually used, and stack space is typically sufficient, so the extra 8 bytes don't cause a real overflow.

--- a/p2p/engine.cc
+++ b/p2p/engine.cc
@@ -1902,11 +1902,15 @@ void Endpoint::cleanup_uds_socket() {
 
 void Endpoint::send_proxy_thread_func() {
   uccl::pin_thread_to_numa(numa_node_);
+  // Use 16-byte buffer to avoid stringop-overflow warning from jring's 16-byte
+  // bulk copy
+  alignas(16) char task_buffer[16];
   UnifiedTask* task;
 
   while (!stop_.load(std::memory_order_acquire)) {
-    if (jring_sc_dequeue_bulk(send_unified_task_ring_, &task, 1, nullptr) ==
-        1) {
+    if (jring_sc_dequeue_bulk(send_unified_task_ring_, task_buffer, 1,
+                              nullptr) == 1) {
+      task = *reinterpret_cast<UnifiedTask**>(task_buffer);
       switch (task->type) {
         case TaskType::SEND_IPC:
           send_ipc(task->conn_id, task->data, task->size);
@@ -1961,11 +1965,15 @@ void Endpoint::send_proxy_thread_func() {
 
 void Endpoint::recv_proxy_thread_func() {
   uccl::pin_thread_to_numa(numa_node_);
+  // Use 16-byte buffer to avoid stringop-overflow warning from jring's 16-byte
+  // bulk copy
+  alignas(16) char task_buffer[16];
   UnifiedTask* task;
 
   while (!stop_.load(std::memory_order_acquire)) {
-    if (jring_sc_dequeue_bulk(recv_unified_task_ring_, &task, 1, nullptr) ==
-        1) {
+    if (jring_sc_dequeue_bulk(recv_unified_task_ring_, task_buffer, 1,
+                              nullptr) == 1) {
+      task = *reinterpret_cast<UnifiedTask**>(task_buffer);
       switch (task->type) {
         case TaskType::RECV_IPC:
           recv_ipc(task->conn_id, task->data, task->size);

